internal/services/appservice/function_app_flex_consumption_resource.go

```diff
@@ -847,47 +1204,134 @@ func (r FunctionAppFlexConsumptionResource) Update() sdk.ResourceFunc {
+			if storageNameIndex := strings.Index(endpoint, "."); storageNameIndex != -1 {
+				deploymentSaName = endpoint[:storageNameIndex]
+			} else {
+				return fmt.Errorf("retrieving storage container endpoint error, the expected format is https://storagename.blob.core.windows.net/containername, the received value is %s", state.DeploymentStorageContainerEndpoint)
+			}
+
+			if deploymentSaKey != "" && deploymentSaName != "" {
+				storageConnStringForFcAppValue = fmt.Sprintf(StorageStringFmt, deploymentSaName, deploymentSaKey, *storageDomainSuffix)
+			}
+
+			// function app backend storage connection string
+			if backendStorageString == "" {
+				backendStorageString = state.StorageAccountName
+			}
+			if !state.StorageAccountUsesMSI {
+				if state.StorageAccountKeyVaultSecretID != "" {
+					backendStorageString = fmt.Sprintf(helpers.StorageStringFmtKV, state.StorageAccountKeyVaultSecretID)
+				} else if state.StorageAccountAccessKey != "" {
+					backendStorageString = fmt.Sprintf(helpers.StorageStringFmt, state.StorageAccountName, state.StorageAccountAccessKey, *storageDomainSuffix)
+				}
+			}
+			if !features.FivePointOh() && (metadata.ResourceData.HasChange("storage_account_name") || metadata.ResourceData.HasChange("storage_account_key_vault_secret_id")) {
+				if state.StorageAccountName == "" && state.StorageAccountKeyVaultSecretID == "" {
+					return fmt.Errorf("the function app storage connection is empty, either set `storage_account_name` or `storage_account_key_vault_secret_id`")
+				}
+			}
+
+			if !features.FivePointOh() && metadata.ResourceData.HasChange("storage_authentication_type") && state.StorageContainerType != "" {
 				storageAuthType := webapps.AuthenticationType(state.StorageAuthType)
 				storageAuth := webapps.FunctionsDeploymentStorageAuthentication{
 					Type: &storageAuthType,
 				}
-				if state.StorageAuthType == string(webapps.AuthenticationTypeStorageAccountConnectionString) {
-					if state.StorageAccessKey == "" {
+				if storageAuthType == webapps.AuthenticationTypeStorageAccountConnectionString {
+					storageConnStringForFCApp = "DEPLOYMENT_STORAGE_CONNECTION_STRING"
+					if deploymentSaKey == "" {
+						return fmt.Errorf("the storage account access key must be specified when using the storage key based access")
+					}
+					storageAuth.StorageAccountConnectionStringName = pointer.To(storageConnStringForFCApp)
+				} else {
+					storageConnStringForFCApp = ""
+					storageConnStringForFcAppValue = ""
+					if storageAuthType == webapps.AuthenticationTypeUserAssignedIdentity {
+						StorageUserAssignedIdentityID := state.StorageUserAssignedIdentityID
+						if StorageUserAssignedIdentityID == "" {
+							return fmt.Errorf("the user assigned identity id must be specified when using the user assigned identity to access the storage account")
+						}
+						storageAuth.UserAssignedIdentityResourceId = &StorageUserAssignedIdentityID
```

Comment:
I know you declare this variable a couple lines above, but for safety, it may be worth using pointer.To
```suggestion
						storageAuth.UserAssignedIdentityResourceId = pointer.To(StorageUserAssignedIdentityID)
```